### ISSUE TITLE
remove t_total from model constructors

### ISSUE THREAD

**Issue by ArneBinder** [ChristophAlt/pytorch-ie#277]
`t_total`, the expected total number of training steps, is required for some models because they are pre-configured to use a training scheduler with warmup. Manually getting the correct value for `t_total` is not straight forward, especially for new users. For the rescue, Pytorch Lightning provides some helper methods for that. We should deprecate that parameter and use instead the following code snipped in the `configure_optimizers` methods:
 
```python
if self.warmup_proportion > 0.0:
    stepping_batches = self.trainer.estimated_stepping_batches
    scheduler = get_linear_schedule_with_warmup(
        optimizer, int(stepping_batches * self.warmup_proportion), stepping_batches
    )
    return [optimizer], [{"scheduler": scheduler, "interval": "step"}]
else:
    return optimizer
```

This effects `TransformerTextClassificationModel` and `TransformerSpanClassificationModel`.
